ouroboros-network-framework/src/Ouroboros/Network/Linger.hsc

```diff
@@ -0,0 +1,50 @@
+#include "HsNet.h"
+
+module Ouroboros.Network.Linger
+  ( StructLinger (..)
+  , reset
+  ) where
+
+import           Control.Monad (when)
+
+import           Foreign.C (CInt)
+import           Foreign.Storable (Storable (..))
+
+import           Network.Socket (Socket)
+import qualified Network.Socket as Socket
+
+
+data StructLinger = StructLinger {
+    -- | Set the linger option on.
+    sl_onoff  :: CInt,
+
+    -- | Linger timeout.
+    sl_linger :: CInt
+  }
+  deriving (Eq, Ord, Show)
+
+instance Storable StructLinger where
+    sizeOf    _ = (#const sizeof(struct linger))
+    alignment _ = alignment (0 :: CInt)
+
+    peek p = do
+        onoff  <- (#peek struct linger, l_onoff) p
+        linger <- (#peek struct linger, l_linger) p
+        return $ StructLinger onoff linger
+
+    poke p (StructLinger onoff linger) = do
+        (#poke struct linger, l_onoff)  p onoff
+        (#poke struct linger, l_linger) p linger
```

Comment:
Can we add a TODO to remove this once we're using the version of the network lib that supports this?